test/tc_profilerecordings_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+func (e *e2e) testCaseProfileRecording([]string) {
+	e.profileRecordingTestCase()
+
+	const exampleRecordingPath = "examples/profilerecording.yaml"
+
+	e.logf("Creating recording")
+	e.kubectl("create", "-f", exampleRecordingPath)
+	defer e.kubectl("delete", "-f", exampleRecordingPath)
+
+	e.logf("Creating test pod")
+	const testPod = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: recording
+  labels:
+    app: alpine
+spec:
+  containers:
+  - image: docker.io/nginx:alpine
+    name: recording
+    resources: {}
+  restartPolicy: Never
+`
+	testPodFile, err := ioutil.TempFile(os.TempDir(), "recording-pod*.yaml")
+	e.Nil(err)
+	defer os.Remove(testPodFile.Name())
+	_, err = testPodFile.Write([]byte(testPod))
+	e.Nil(err)
+	err = testPodFile.Close()
+	e.Nil(err)
+	e.kubectl("create", "-f", testPodFile.Name())
+
+	e.logf("Waiting for test pod to be initialized")
+	e.waitFor("condition=initialized", "pod", "recording")
+
+	output := e.kubectl("get", "pod", "recording")
+	for strings.Contains(output, "ContainerCreating") {
+		output = e.kubectl("get", "pod", "recording")
+	}
```

Comment:
why not wait for the pod to be ready?

Reply:
Yeah this makes more sense, I changed the test logic which should also simplify the test.